core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java

```diff
@@ -29,23 +27,34 @@ public Collection<Pathfinding.Range> apply(SignalingRoute reservationRoute) {
      * Returns the sections of the given route that can't be used by the given rolling stock
      * because it needs electrified tracks and isn't compatible with the catenaries in some range
      */
-    private static Set<Pathfinding.Range> getBlockedRanges(RollingStock stock, ReservationRoute route) {
+    static Set<Pathfinding.Range> getBlockedRanges(RollingStock stock, ReservationRoute route) {
         if (!stock.isElectricOnly())
             return Set.of();
 
         var res = new HashSet<Pathfinding.Range>();
         double offset = 0;
         for (var range : route.getTrackRanges()) {
             var voltages = range.getCatenaryVoltages();
+            var deadSections = range.getDeadSections();
             for (var section : voltages.asMapOfRanges().entrySet()) {
                 var interval = section.getKey();
                 if (Math.abs(interval.lowerEndpoint() - interval.upperEndpoint()) < 1e-5)
                     continue;
                 if (!stock.getModeNames().contains(section.getValue())) {
-                    res.add(new Pathfinding.Range(
-                            offset + interval.lowerEndpoint(),
-                            offset + interval.upperEndpoint())
-                    );
+                    List<Range<Double>> blockingRanges = new ArrayList<>();
+                    if (deadSections.isEmpty()) {
+                        blockingRanges.add(interval);
```

Comment:
Do we need to handle this case separately? It should be covered by the general case (assuming `deadSections.complement` behaves reasonably on an empty range)

Reply:
Indeed, we don't, we did that to prevent some computations, we it was just premature optimization. After measuring performances, it turns out it doesn't change anything